sub_circuit_identification/src/write_verilog_lef.py

```diff
@@ -108,6 +110,61 @@ def map_pins(self, a, b):
             print("unmatched ports found")
             return 0
 
+class WriteSpice:
+    """ write hierarchical verilog file """
+
+    def __init__(self, circuit_graph, circuit_name, inout_pin_names,subckt_list):
+        self.circuit_graph = circuit_graph
+        self.circuit_name = circuit_name
+        self.inout_pins = inout_pin_names
+        self.pins = inout_pin_names
+        self.subckt_list = subckt_list
+
+    def check_ports_match(self,port,subckt):
+        for members in self.subckt_list:
+            if members["name"]==subckt and port in members["ports"]:
+                return 1
+
+    def print_subckt(self, fp):
+        logging.info("Writing module : %s", self.circuit_name)
+        fp.write("\n.subckt " + self.circuit_name + " ")
+        fp.write(' '.join(self.pins))
+
+        for node, attr in self.circuit_graph.nodes(data=True):
+            if 'source' in attr['inst_type']:
+                logging.info("Skipping source nodes : %s", node)
+                continue
+            if 'net' not in attr['inst_type']:
+                logging.info("Writing node: %s", attr['inst_type'])
+                fp.write("\n" + node + ' '+ attr['real_inst_type'] + ' ')
```

Comment:
instance name needs to go after pin names

Reply:
Yes, I will fix this.

Reply:
Thank you !